Did library support cryptographic key types

Kia82 opened the pull request:
This Pull Request focuses on several significant updates and refinements to the codebase, particularly targeting DID-related functionality, configuration adjustments, and cleanup.

Main Changes:
- Global .env and Configuration Updates:
- Added registry paths for DID-related functionality in .env files.
- Adjusted references to environment variables in code (e.g., REACT_APP_API_URL).

Enhancements in DID Library:
- Introduced a centralized registry system for storing and managing DIDDocument instances with added persistence.
- Improvements in KeyDID and DID handling, including added methods for generating/storing DIDs and verification methods.
- Removed unused dependencies (e.g., bs58, uuid).

Refactoring:
- Code clean-ups in KeyDID handlers and helper utils.
- Simplified formatting and key generation logic for DID-related methods.
- Reorganized files and added data structure logic under did/registry for proper separation.

Application/Frontend Updates:
- Updated frontend API-related configurations for better environment management.

Other:
- Minor code formatting (e.g., spacing, linting adjustments).
- Renamed/moved JSON-LD example files for credential resources.

Commit:
added core functions and implementation

Commit:
redesign of the handler to support dynamic did structures, added did definitions

Commit:
Merge branch 'dev' into did-library-core-implementation

Commit:
improved support for handler, commented key_utils due to failures

Commit:
skeleton code added

Commit:
implemented core functions, add test data

Commit:
connected front end to backend

Commit:
added support for dynamic key collection

Commit:
fix merge conflict

Commit:
refactor: code readability and improved error handeling

Commit:
space removal

Commit:
Added persistant did storage solution, created helper functions for did methods as well as moved them to key_utils class

Commit:
fixed up code logic, added todos, add path resolver function

Commit:
conneted did document file storage solution to app

AlirezaKondori commented on .gitignore:
```diff
@@ -8,7 +8,7 @@ target
 /dataSources/
 /dataSources.local.xml
-
+/crates/did_library/resources
```
Why are we removing this folder from the repo?

Kia82 replied:
Because it shouldn't be in the remote repo

AlirezaKondori commented on crates/api_server/src/main.rs:
```diff
@@ -65,19 +65,31 @@ struct IssueCredentialRequest {
 #[derive(Deserialize)]
-struct CreateDIDRequest {
-}
+struct CreateDIDRequest {}
```
Empty struct?

Kia82 replied:
It's empty for now as we do not have front end support to modify our did. Once we do, we can add the appropriate fields to the request handler itself.

AlirezaKondori commented on crates/api_server/src/main.rs:
```diff
@@ -65,19 +65,31 @@ struct IssueCredentialRequest {
 async fn create_did_handler(req: web::Json<CreateDIDRequest>) -> impl Responder {
+
+    // Path to the registry file
+    let registry_path = "../did_library/resources/did_registry.json";
```
I think this path should be protected. Could you put into the .env?

Kia82 replied:
Will do it

AlirezaKondori commented on crates/api_server/src/main.rs:
```diff
@@ -93,11 +105,18 @@ async fn create_did_handler(req: web::Json<CreateDIDRequest>) -> impl Responder
+    //TODO: Dynamically load and configure path for registry initialization (from .env file).
+    // Alireza if you see this, you are going to laugh at me for not doing this.
+    did_library::did::registry::init_registry(Some(
+        "../did_library/resources/did_registry.json".to_string(),
```
Protect the path here also

AlirezaKondori commented on crates/did_library/src/did/core/key_utils.rs:
```diff
@@ -27,6 +28,57 @@ pub struct PublicKey {
 
+/// Hash a JWK into three fields: key_type, public_key, and private_key.
+/// Note all key,value pairs are strings
+pub fn hash_jwk(jwk: &serde_json::Value) -> Result<HashMap<String, String>, String> {
```
Add some comments to this function pls

Kia82 replied:
Will do

AlirezaKondori commented on crates/did_library/src/did/methods/ethr/handler.rs:
```diff
@@ -5,22 +5,25 @@ use crate::DIDDocument;
-	
-}
\ No newline at end of file
+    fn update_did(
```
I think you have this function implemented in `traits.rs`. Look into pulling one of them into the other file for less duplication if they're going to be identical. Otherwise, try to name them slightly differently. Maybe `update_did_handler` and `update_did_traits`

Kia82 replied:
The traits.rs is an trait class that each did method is pulling from. Each method is implementing this trait and has it's own unqiue implementation. We must use the same name to override the function or it will lead to compile time issues

Commit:
removed unneeded files

Commit:
Refactor DID registry path to use environment variable

Replaced hardcoded DID registry path with an environment variable for better configurability. Updated `.env` file and related code to reflect this change. Also added TODOs for user input handling and improved `CreateDIDRequest` structure for future implementation.

Commit:
fix merge conflict
